tachyon/math/base/big_int.h

```diff
@@ -681,10 +681,10 @@ struct ALIGNAS(internal::LimbsAlignment(N)) BigInt {
   }
 
   template <bool ModulusHasSpareBit>
-  constexpr BigInt MontgomeryInverse(const BigInt& modulus,
-                                     const BigInt& r2) const {
-    // See https://github.com/kroma-network/tachyon/issues/76
-    CHECK(!IsZero());
+  [[nodiscard]] constexpr bool MontgomeryInverse(const BigInt& modulus,
+                                                 const BigInt& r2,
+                                                 BigInt& output) const {
+    if (UNLIKELY(IsZero())) return false;
```

Comment:
Could you add logging here?

```suggestion
    if (UNLIKELY(IsZero())) {
      LOG(ERROR) << "Inverse of zero attempted";
      return false;
    }
```